app/controllers/merge_controller.rb

```diff
@@ -29,7 +28,11 @@ def merge!
       redirect_to @dest
     when Modes::REVIEW
       @merge_request.send("#{@decision}_by!".to_sym, current_user)
-      redirect_to @merge_request.source
+      redirect_to @merge_request.dest, notice: "Merge request #{@decision}"
+    when Modes::REQUEST
+      mr = MergeRequest.create(user: current_user, source: @source, dest: @dest)
```

Comment:
minor: you might consider using ``` create! ``` here so it doesn't silently fail and pass nil into the mailer.

Reply:
+1